v0.33.0 Release notes (#4886)
graphprotocol/graph-node: NEWS.md

- Added a '`paused`' field to Index Node API, a boolean indicating the subgraph’s pause status. [(#4779)](https://github.com/graphprotocol/graph-node/pull/4779)
- Proof of Indexing logs now include block number [(#4798)](https://github.com/graphprotocol/graph-node/pull/4798)
- `subgraph_features` table now tracks details about handlers used in a subgraph [(#4820)](https://github.com/graphprotocol/graph-node/pull/4820)
- Configurable SSL for Postgres in Dockerfile -  ssl-mode for Postgres can now be configured via the connection string when deploying through Docker, offering enhanced flexibility in database security settings.[(#4840)](https://github.com/graphprotocol/graph-node/pull/4840)
- Introspection Schema Update - The introspection schema has been updated to align with the October 2021 GraphQL specification update.[(#4676)](https://github.com/graphprotocol/graph-node/pull/4676)
- `trace_id` Added to Substreams Logger [(#4868)](https://github.com/graphprotocol/graph-node/pull/4868)

> - New apiVersion for Mapping Validation - The latest apiVersion 0.0.8 now validates that fields set in entities from the mappings are actually defined in the schema. Subgraphs using this new API version will fail if they try to set undefined schema fields in the mappings. [(#4894)](https://github.com/graphprotocol/graph-node/pull/4894)

Author: > The latest apiVersion 0.0.8 validates that fields set in entities from the mappings are actually defined in the schema. This fixes a source of non-deterministic PoI. Subgraphs using this new API version will fail if they try to set undefined schema fields in the mappings. Its strongly recommended updating to 0.0.8 to avoid these issues.

Going with this

- Substreams Block Ingestor Support - Added the ability to run a pure substreams chain by introducing a block ingestor for substreams-only chains. This feature allows users to run a chain with just a single substreams endpoint, enhancing support beyond RPC and firehose. Prior to this, a pure substreams chain couldn’t be synced.[(#4839)](https://github.com/graphprotocol/graph-node/pull/4839)

### Bug fixes

- Fix for rewinding dynamic data source - Resolved an issue where a rewind would fail to properly remove dynamic data sources when using `graphman rewind`. This has been fixed to ensure correct behavior.[(#4810)](https://github.com/graphprotocol/graph-node/pull/4810)
- Improved Deployment Reliability with Retry Mechanism - A retry feature has been added to the block_pointer_from_number function to enhance the robustness of subgraph deployments. This resolves occasional failures encountered during deployment processes.[(#4812)](https://github.com/graphprotocol/graph-node/pull/4812)

> - Fixed Cross-Shard Grafting Issue - Addressed a bug that prevented cross-shard grafting from starting, causing the copy operation to stall at 0% progress. This issue occurred when a new shard was added after the primary shard had already been configured. The fix ensures that foreign tables and schemas are correctly set up in new shards.[(#4845)](https://github.com/graphprotocol/graph-node/pull/4845)

Author: noted, changing the wording

- Fixed a Full-text search regression - Reverted a previous commit (ad1c6ea) that inadvertently limited the number of populated search indexes per entity.[(#4808)](https://github.com/graphprotocol/graph-node/pull/4808)
- Attestable Error for Nested Child Filters - Nested child filter queries now return an attestable `ChildFilterNestingNotSupportedError`, improving error reporting for users.[(#4828)](https://github.com/graphprotocol/graph-node/pull/4828)

### Graphman

> - **Index on prefixed fields** - The graphman index create command now correctly indexes prefixed fields of type String and Bytes, making combined indexes more query-efficient.

Reviewer: I am not 100% sure that the behavior here is correct: IIRC, the logic is that if the field is a reference to an entity, we do _not_ use `left(..)` in queries, but `graphman index create` will create an index on `left(..)` when it should create one on the column directly

Author: Got it, I'll go with something like this

"Index on Prefixed Fields - The graphman index create command now correctly indexes prefixed fields of type String and Bytes for more query-efficient combined indexes. Note: For fields that are references to entities, the behavior may differ. The command may create an index using left(..) when it should index the column directly.

## Unreleased

## v0.33.0

### What's New

> - **Arweave file data sources** - Arweace file data sources allow subgraph developers to query offchain information from Arweave directly in mappings.[(#4789)](https://github.com/graphprotocol/graph-node/pull/4789)

Reviewer 2: ```suggestion
- **Arweave file data sources** - Arweave file data sources allow subgraph developers to query offchain information from Arweave from within the subgraph mappings.[(#4789)](https://github.com/graphprotocol/graph-node/pull/4789)
```

Reviewer 3: I think "access" rather than "query" might be clearer here

> - **Major performance boost for substreams-based subgraphs** - Significant performance improvements have been achieved for substreams-based subgraphs by moving substreams processing to the block stream.[(#4851)](https://github.com/graphprotocol/graph-node/pull/4851)

Reviewer 3: @mangas do you have an estimate of the size of the boost?

> - **Polling block handler** - A new block handler filter `polling` for `ethereum` datasources which enables subgraph developers to run a block handler at defined block intervals. This is useful for use cases such as taking snapshots at specific block intervals.[(#4725)](https://github.com/graphprotocol/graph-node/pull/4725)

Reviewer 2: ```suggestion
- **Polling block handler** - A new block handler filter `polling` for `ethereum` datasources which enables subgraph developers to run a block handler at defined block intervals. This is useful for use cases such as taking taking periodic snapshots of the contract state.[(#4725)](https://github.com/graphprotocol/graph-node/pull/4725)
```

> - **Initialization handler** - A new block handler filter `once` for `ethereum` datasources which enables subgraph developers to create a handle which will be called only once before all other handlers run. This configuration allows the subgraph to use the handler as an initialization handler, performing specific tasks at the start of indexing. [(#4725)](https://github.com/graphprotocol/graph-node/pull/4725)

Reviewer 2: ```suggestion
- **Initialization handler** - A new block handler filter `once` for `ethereum` data sources which enables subgraph developers to create a handle which will be called only once before all other handlers run. This configuration allows the subgraph to use the handler as an initialization handler, performing specific tasks at the start of indexing. [(#4725)](https://github.com/graphprotocol/graph-node/pull/4725)
```

- `graph-node` version in index node API - The Index Node API now features a new query, Version, which can be used to query the current graph-node version and commit. [(#4852)](https://github.com/graphprotocol/graph-node/pull/4852)
- Added subgraph status to index node API [(#4779)](https://github.com/graphprotocol/graph-node/pull/4779)
- Proof of Indexing logs now include block number [(#4798)](https://github.com/graphprotocol/graph-node/pull/4798)
- `subgraph_features` table now tracks details about handlers used in a subgraph [(#4820)](https://github.com/graphprotocol/graph-node/pull/4820)
- Configurable SSL for Postgres in Dockerfile -  ssl-mode for Postgres can now be configured via the connection string when deploying through Docker, offering enhanced flexibility in database security settings.[(#4840)](https://github.com/graphprotocol/graph-node/pull/4840)
- Introspection Schema Update - The introspection schema has been updated to align with the October 2021 GraphQL specification update.[(#4676)](https://github.com/graphprotocol/graph-node/pull/4676)
- `trace_id` Added to Substreams Logger [(#4868)](https://github.com/graphprotocol/graph-node/pull/4868)

> * Add substreams block ingestor by @mangas in https://github.com/graphprotocol/graph-node/pull/4839

Reviewer 3: Do we need more notes here?

* store: Map other shards into a new shard on startup by @lutter in https://github.com/graphprotocol/graph-node/pull/4845

### Breaking changes

> - Validate field exists in schema before setting entity by @incrypto32 in https://github.com/graphprotocol/graph-node/pull/4807

Reviewer 3: Changing with the discussed fix 🙏
